asynch/pool.py

```diff
@@ -188,7 +209,8 @@ async def connection(self) -> AsyncIterator[Connection]:
 
         async with self._sem:
             async with self._lock:
-                if not self._free_connections:
+                minsize = self.minsize
+                if self.free_connections < minsize:
```

Comment:
Yep, connections are to be initiated not when the `Pool.minsize` property reaches zero, but when this number goes below this value.

Reply:
Rethought.